### PR TITLE
Check positive dilation in `F.convolution_2d`

### LINKED ISSUE #7304: CUDNN_STATUS_BAD_PARAM raised when dilate=0 in L.Convolution2D
The sample code I tried:
```
import numpy as np
import chainer
import chainer.links as L

# cpu
link = L.Convolution2D(in_channels=1, out_channels=64, ksize=3, stride=(1, 1), pad=(0, 0), nobias=False, dilate=(0, 0), groups=1)
data = np.random.random((1, 1, 3, 80)).astype(np.float32)
o = link(data)

print('cpu:', o.shape)

print('===================')
chainer.print_runtime_info()
print('===================')
# gpu
gpu_id = 0
to_device = chainer.dataset.convert.to_device
chainer.cuda.get_device_from_id(gpu_id).use()
link.to_gpu()
data = to_device(gpu_id, data)
o = link(data)
print('gpu:', o.shape)
```

result:
```
cpu: (1, 64, 3, 80)
===================
Platform: Linux-4.4.0-142-generic-x86_64-with-debian-stretch-sid
Chainer: 6.0.0
NumPy: 1.16.3
CuPy:
  CuPy Version          : 6.0.0
  CUDA Root             : /usr/local/cuda-9.0
  CUDA Build Version    : 9000
  CUDA Driver Version   : 10010
  CUDA Runtime Version  : 9000
  cuDNN Build Version   : 7402
  cuDNN Version         : 7402
  NCCL Build Version    : 2307
  NCCL Runtime Version  : 2307
iDeep: Not Available
===================
Traceback (most recent call last):
  File "test_vgg.py", line 21, in <module>
    o = link(data)
  File "/env/.pyenv/versions/speech_3.6/lib/python3.6/site-packages/chainer/link.py", line 294, in __call__
    out = forward(*args, **kwargs)
  File "/env/.pyenv/versions/speech_3.6/lib/python3.6/site-packages/chainer/links/connection/convolution_2d.py", line 184, in forward
    groups=self.groups)
  File "/env/.pyenv/versions/speech_3.6/lib/python3.6/site-packages/chainer/functions/connection/convolution_2d.py", line 582, in convolution_2d
    y, = fnode.apply(args)
  File "/env/.pyenv/versions/speech_3.6/lib/python3.6/site-packages/chainer/function_node.py", line 321, in apply
    outputs = self.forward(in_data)
  File "/env/.pyenv/versions/speech_3.6/lib/python3.6/site-packages/chainer/function_node.py", line 512, in forward
    return self.forward_gpu(inputs)
  File "/env/.pyenv/versions/speech_3.6/lib/python3.6/site-packages/chainer/functions/connection/convolution_2d.py", line 185, in forward_gpu
    return self._forward_cudnn(x, W, b, y)
  File "/env/.pyenv/versions/speech_3.6/lib/python3.6/site-packages/chainer/functions/connection/convolution_2d.py", line 246, in _forward_cudnn
    auto_tune=auto_tune, tensor_core=tensor_core)
  File "cupy/cudnn.pyx", line 1545, in cupy.cudnn.convolution_forward
  File "cupy/cudnn.pyx", line 216, in cupy.cudnn._create_convolution_descriptor
  File "cupy/cuda/cudnn.pyx", line 1078, in cupy.cuda.cudnn.setConvolution2dDescriptor_v5
  File "cupy/cuda/cudnn.pyx", line 712, in cupy.cuda.cudnn.check_status
cupy.cuda.cudnn.CuDNNError: CUDNN_STATUS_BAD_PARAM
```

### PR DESCRIPTION
**Pull request by takagi**
Close #7304.

This PR make `F.convolution_2d` check if the supplied dilation is positive integer.

### CODE REVIEW

File: tests/chainer_tests/functions_tests/connection_tests/test_convolution_2d.py [hunk 1]
```diff
@@ -300,4 +300,47 @@ def test_1(self):
...
+                self.check_invalid_dilation(x, w)
+
+    def test_invalid_dilation_cpu_ideep(self):
+        x = numpy.ones(self.x_shape, numpy.float32)
+        w = numpy.ones(self.w_shape, numpy.float32)
+        with self.assertRaises(ValueError):
+            with chainer.using_config('use_ideep', 'always'):
+                self.check_invalid_dilation(x, w)
+
+    def test_invalid_dilation_gpu(self):
```
Comment (toslunar) [inline tests/chainer_tests/functions_tests/connection_tests/test_convolution_2d.py]: Could you mark GPU tests?  Another option is to use `chainer.testing.inject_backend_tests`.

Comment (takagi) [reply tests/chainer_tests/functions_tests/connection_tests/test_convolution_2d.py]: Added `@gpu`, `@cudnn` and `@ideep` marks!

I suppose that `chainer.testing.inject_backend_tests` needs to be used with `chainer.testing.FunctionTestCase`, which does not support cases expecting an error raised as it is. Right?

Comment (toslunar) [reply tests/chainer_tests/functions_tests/connection_tests/test_convolution_2d.py]: `@attr.gpu`, etc?

Comment (takagi) [reply tests/chainer_tests/functions_tests/connection_tests/test_convolution_2d.py]: Yes...

**Comment by toslunar** [comment]
CIs, test this please.

**Pull request merged**